Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -310,4 +323,33 @@ void WebCookieJar::setCookieAsync(WebCore::Document& document, const URL& url, c
     WebProcess::singleton().ensureNetworkProcessConnection().connection().sendWithAsyncReply(Messages::NetworkConnectionToWebProcess::SetCookieFromDOMAsync(document.firstPartyForCookies(), sameSiteInfo, url, frameID, pageID, applyTrackingPreventionInNetworkProcess, shouldRelaxThirdPartyCookieBlocking(webFrame), cookie), WTFMove(completionHandler));
 }
 
+#if HAVE(COOKIE_CHANGE_LISTENER_API)
+void WebCookieJar::addChangeListener(const String& host, const WebCore::CookieChangeListener& cookieChangeListener)
+{
+    auto it = m_changeListeners.find(host);
+    if (it != m_changeListeners.end() && it->value.contains(cookieChangeListener))
+        return;
+
+    m_changeListeners.ensure(host, [] {
+        return WeakHashSet<CookieChangeListener> { };
+    }).iterator->value.add(cookieChangeListener);
+
```

Comment:
```
auto& listenersForHost = m_changeListeners.add(host, WeakHashSet<CookieChangeListener> { });
auto addResult = listenersForHost.add(cookieChangeListener);
if (listenersForHost.size() != 1 || !addResult.isNewEntry)
    return;
```
is what I would do.

Reply:
If there are already listeners for this host in `m_changeListeners`, wouldn't 
`auto& listenersForHost = m_changeListeners.add(host, WeakHashSet<CookieChangeListener> { });`
remove them because it will replace the WeakHashSet which contains those listeners with an empty WeakHashSet?

Reply:
No, add() doesn’t replace, unlike the set() function.